java/shared/src/test/java/com/squareup/subzero/shared/SubzeroUtilsTest.java

```diff
@@ -526,4 +610,34 @@ private TxOutput testOutput(long amount, Destination destination) {
             .setIndex(2))
         .build();
   }
+
+  private ECKey newMockPubKey() {
+    return new ECKey();
+  }
+
+  private Signature createValidMockSignature(byte[] hash, ECKey pubkey) {
+    Sha256Hash sha256Hash = Sha256Hash.wrap(hash);
+    ECKey.ECDSASignature ecdsaSignature = pubkey.sign(sha256Hash);
+    byte[] signatureBytes = ecdsaSignature.encodeToDER();
+    return Signature.newBuilder()
+        .setHash(ByteString.copyFrom(hash))
+        .setDer(ByteString.copyFrom(signatureBytes))
+        .build();
+  }
+
+  private byte[] generateMockHash(String inputData) {
+    try {
+      MessageDigest digest = MessageDigest.getInstance("SHA-256");
+      return digest.digest(inputData.getBytes(StandardCharsets.UTF_8));
+    } catch (NoSuchAlgorithmException e) {
+      throw new RuntimeException("SHA-256 not found", e);
+    }
+  }
+
+  private Signature createInvalidMockSignature() {
+    return Signature.newBuilder()
+        .setHash(ByteString.copyFrom(new byte[]{0}))
+        .setDer(ByteString.copyFrom(new byte[]{0}))
+        .build();
+  }
```

Comment:
naming nit: maybe call this `createInvalidEmptySignature()`?

Also could be interesting to have a test case where we flip a bit in a valid signature, you will probably want a separate test helper for that.